File: iosApp/iosApp/Services/Etherscan/EtherscanServiceAssembler.swift
// Created by web3d4v on 05/08/2022.
// Copyright (c) 2022 Sons Of Crypto.
// SPDX-License-Identifier: MIT

import web3walletcore

final class EtherscanServiceAssembler: AssemblerComponent {
    
    func register(to registry: AssemblerRegistry) {
        
        registry.register(scope: .singleton) { resolver -> EtherScanService in
            DefaultEtherScanService(
                store: KeyValueStore(name: "\(EtherScanService.self)")
            )
        }
    }
}

File: androidApp/src/main/java/com/sonsofcrypto/web3wallet/android/services/EtherScanServiceAssembler.kt
package com.sonsofcrypto.web3wallet.android.services

import com.sonsofcrypto.web3lib.keyValueStore.KeyValueStore
import com.sonsofcrypto.web3wallet.android.common.AssemblerComponent
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistry
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistryScope
import com.sonsofcrypto.web3walletcore.services.etherScan.DefaultEtherScanService
import com.sonsofcrypto.web3walletcore.services.etherScan.EtherScanService
import smartadapter.internal.extension.name

class EtherScanServiceAssembler: AssemblerComponent {

    override fun register(to: AssemblerRegistry) {

        to.register(EtherScanService::class.name, AssemblerRegistryScope.SINGLETON) {
            DefaultEtherScanService(
                it.resolve("${KeyValueStore::class.name}.EtherScanService"),
            )
        }
    }
}